lib/content-info/index.js

```diff
@@ -0,0 +1,381 @@
+    *
+    * @private
+    * @param {String} dispositionHeader - Content-disposition Header from the response
+    * @returns {?String} Returns file name from content disposition header if present
+    */
+    getFileNameFromDispositionHeader = function (dispositionHeader) {
+        if (!dispositionHeader) {
+            return;
+        }
+
+        var encodedFileName,
+            fileName;
+
+        // Get filename* value from the dispositionHeader
+        encodedFileName = encodedFileNameRegex.exec(dispositionHeader);
+
+        if (encodedFileName) {
+            fileName = decodeFileName(encodedFileName[Number_2], encodedFileName[Number_1]);
+        }
+
+        // If filename* is not present or unparseable, then we are checking for filename in header
+        if (!fileName) {
+            fileName = fileNameRegex.exec(dispositionHeader);
+            fileName && (fileName = fileName[Number_1]);
+            if (fileName && fileName[Number_0] === doubleQuotes) {
+                // remove quotes and escapes
+                fileName = fileName
+                    .substr(Number_1, fileName.length - Number_2)
+                    .replace(quotedPairRegex, TOKEN_$1);
+            }
+        }
+
+        return fileName;
+    };
+
+
+module.exports = {
+
+    /**
+    * From response object, mime info and filename of the response content is taken
+    *
+    * @param {Response} response - response instance
+    *
+    * @returns {ResponseContentInfo} - Return contentInfo of the response
+    */
+    contentInfo: function (response) {
+        var contentType = response.headers.get(CONTENT_TYPE),
+            contentDisposition = response.headers.get(CONTENT_DISPOSITION),
+            mimeInfo = getMimeInfo(contentType, response.stream || response.body),
+            fileName = getFileNameFromDispositionHeader(contentDisposition),
+            extension = mimeInfo.extension,
+
+            /**
+             * @typedef ResponseContentInfo
+             *
+             * @property {String} mimeType sanitized mime type
+             * @property {String} mimeFormat format for the identified mime type
+             * @property {String} charset the normalized character set
+             * @property {String} extension extension identified from the mime type
+             * @property {String} fileName file name extracted from disposition header
+             */
+            contentInfo = {};
+
+
+        // if file name is not present in the content disposition headers, use a default file name
+        if (!fileName) {
+            // add extension to default if present
+            extension ? (fileName = DEFAULT_RESPONSE_FILENAME + DOT + extension) :
+                (fileName = DEFAULT_RESPONSE_FILENAME);
+        }
+
+        // create a compacted list of content info from mime info and file name
+        mimeInfo.mimeType && (contentInfo.mimeType = mimeInfo.mimeType);
```

Comment:
@kunagpal is there a cleaner way to assign non empty values to new objects?

Reply:
Easier to read yes, but not as performant as this.